anesthetic/labelled_pandas.py

```diff
@@ -46,22 +48,35 @@ def cmp(x, y):
 
     results.sort(key=cmp_to_key(cmp))
 
-    for s in results:
+    for s, l in results:
         if s is not None:
+            if hasattr(s, "name"):
+                try:
+                    s.name = l[s.name]
+                except Exception:
+                    pass
             return s
     raise errors[-1]
 
 
 class _LocIndexer(_LocIndexer_):
     def __getitem__(self, key):
-        return ac([_LocIndexer_("loc", self.obj.drop_labels(i)).__getitem__
-                   for i in self.obj._all_axes()] + [super().__getitem__], key)
+        return ac([(_LocIndexer_("loc",
+                                 super(_LabelledObject,
+                                       self.obj.drop_labels(i))
+                                 ).__getitem__,
+                    self.obj.get_labels_map(i))
+                   for i in self.obj._all_axes()], key)
```

Comment:
There are several changes to all of the indexers/accessors:
1) `_all_axes` now returns `None` as well, to capture the extra edge case (which was previously tagged on as `+[...]`
2) crucially we return the `super` object in the `Indexers` (this is one of those, 'how did that ever work?' questions
3) We return a list of (function, labels_map) pairs, rather than just a list of functions.